### ISSUE TITLE
Exception if /tmp/bosco/.clusterlist doesn't exist

### ISSUE THREAD

**Issue by LincolnBryant** [vc3-project/vc3-remote-manager#7]
@manglano gets an error when creating this cluster
```
2019-01-31 13:27:55,444 (UTC) [ ERROR ] autopyfactory.confighandler config.py:61 _run(): setting configuration for AuthManager failed. Will not proceed with threads configuration.
2019-02-07 16:13:52,354 (UTC) [ ERROR ] autopyfactory remotemanager.py:73 _checktarget(): Exception during bosco remote installation. 
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/autopyfactory/remotemanager.py", line 65, in _checktarget
    clusters = bosco.get_clusters()
  File "/usr/lib/python2.7/site-packages/vc3remotemanager/bosco.py", line 316, in get_clusters
    with open(self.clusterlist, 'r') as f:
IOError: [Errno 2] No such file or directory: '/tmp/bosco/.clusterlist'
```

We need to change bosco.py such that it stats self.clusterlist to see if it exists, and if it doesn't, create it.

**Comment by ghost** [comment]
o.k., now this checks the mode on open and revises it to create the file if it doesn't exist; and, it uses stat to check if the directory exists. I also found and fixed another possible error in add_clusters method. - Dave

**Comment by LincolnBryant** [comment]
Fixed by #8
